### ISSUE TITLE
wp-env: No longer working with a local checkout of WordPress

### ISSUE THREAD

**Issue by talldan** [WordPress/gutenberg#30655]
## Description
wp-env was previously useful for core development. It was possible to checkout a local version of WordPress (for example, from the wordpress-develop repo) and change the core setting in the wp-env config to point to the build folder.

That seems to have stopped working recently. I'm now seeing the `wp-env start` command fail with:
```
Error: This does not seem to be a WordPress installation.
Pass --path=`path/to/wordpress` or run `wp core download`.
```

@tellthemachines also mentioned that the environment stops working whenever running `npm run build` in the wordpress-develop repo, which is something that also happened to me, so I'm not sure if that's related. I had a look over changes to the core build process and couldn't see a whole lot that's different.

## Step-by-step reproduction instructions
1. Clone https://github.com/WordPress/wordpress-develop
2. Run `npm i` and `npm run build` in the cloned folder
3. Back in gutenberg add a `.wp-env.override.json` file with a core setting that points to the build folder:
```json
{
	"core": "../wordpress-develop/build"
}
```
4. Run `npm run wp-env start`
5. Try accessing the dev environment in a web browser, the first time for me it worked ok
6. In wordress-develop run `npm run build` again.
7. Try accessing the dev environment in a web browser

## Expected behaviour
The dev environment still works

## Actual behaviour
The dev environment is inaccessible. Attempting to restart it results in the above error message

**Comment by talldan** [comment]
Not seeing the same error any longer. Though the WordPress core `npm run build` command does still seem to cause a problem.

`npm run wp-env start` completes ok, but there's a `403 Forbidden: You don't have permission to access this resource` error when trying to access `localhost:8888/wp-admin`.

**Comment by talldan** [comment]
An update to this, I managed to get this working some time ago. The issue seems to be that wp-env doesn't generate a wp-config.php file for local versions of WordPress. I believe it needs to create its own wp-config that has the `getenv_docker` function.

My workaround was to copy the correct wp-config. I did so by first starting up wp-env in its default configuration. It'll fetch WordPress from the git repo and install it into a subfolder of the `~/.wp-env` folder. I copied the wp-config from that sub folder into the `src` and/or `build` folders of the local repo. Then configured my `.wp-env.override.json` to point to the src or build folder of the local repo and destroyed and restarted wp-env.
